extras/indexingExample/src/main/java/InferenceExamples.java

```diff
@@ -0,0 +1,592 @@
+
+	       // MongoDBIndexingConfigBuilder builder = MongoIndexingConfiguration.builder()
+	       //     .setUseMockMongo(USE_MOCK).setUseInference(USE_INFER).setAuths("U");
+	        MongoDBIndexingConfigBuilder builder = MongoIndexingConfiguration.builder()
+		            .setUseMockMongo(USE_EMBEDDED_MONGO).setUseInference(true).setAuths("U");
+	        
+	        if (USE_EMBEDDED_MONGO) {
+	            final MongoClient c = MockMongoFactory.newFactory().newMongoClient();
+	            final ServerAddress address = c.getAddress();
+	            final String url = address.getHost();
+	            final String port = Integer.toString(address.getPort());
+	            c.close();
+	            builder.setMongoHost(url).setMongoPort(port);
+	        } else {
+	            // User name and password must be filled in:
+	            builder = builder.setMongoUser(MongoUserName)
+	                             .setMongoPassword(MongoUserPassword)
+	                             .setMongoHost(MONGO_INSTANCE_URL)
+	                             .setMongoPort(MONGO_INSTANCE_PORT);
+	        }
+
+	        return builder.setMongoDBName(MONGO_DB)
+	               .setMongoCollectionPrefix(MONGO_COLL_PREFIX)
+	               .setUseMongoFreetextIndex(true)
+	               .setMongoFreeTextPredicates(RDFS.LABEL.stringValue()).build();
+
+	    }
+
+
+	    public static void testPropertyChainInferenceAltRepresentation(final SailRepositoryConnection conn, final Sail sail) throws MalformedQueryException, RepositoryException,
+	    UpdateExecutionException, QueryEvaluationException, TupleQueryResultHandlerException, InferenceEngineException {
+
+	        // Add data
+	        String query = "INSERT DATA\n"//
+	                + "{ GRAPH <http://updated/test> {\n"//
+	                + "  <urn:jenGreatGranMother> <urn:Motherof> <urn:jenGranMother> . "
+	                + "  <urn:jenGranMother> <urn:isChildOf> <urn:jenGreatGranMother> . "
+	                + "  <urn:jenGranMother> <urn:Motherof> <urn:jenMother> . "
+	                + "  <urn:jenMother> <urn:isChildOf> <urn:jenGranMother> . "
+	                + " <urn:jenMother> <urn:Motherof> <urn:jen> . "
+	                + "  <urn:jen> <urn:isChildOf> <urn:jenMother> . "
+	                + " <urn:jen> <urn:Motherof> <urn:jenDaughter> .  }}";
+
+	        log.info("Performing Query");
+
+	        Update update = conn.prepareUpdate(QueryLanguage.SPARQL, query);
+	        update.execute();
+
+	        query = "select ?p { GRAPH <http://updated/test> {?s <urn:Motherof>/<urn:Motherof> ?p}}";
+	        CountingResultHandler resultHandler = new CountingResultHandler();
+	        TupleQuery tupleQuery = conn.prepareTupleQuery(QueryLanguage.SPARQL, query);
+	        tupleQuery.evaluate(resultHandler);
+	        log.info("Result count : " + resultHandler.getCount());
+
+
+	        // try adding a property chain and querying for it
+	        query = "INSERT DATA\n"//
+	                + "{ GRAPH <http://updated/test> {\n"//
```

Comment:
Is there a cleaner sparql syntax for this?  You don't need to define the blank nodes, and there's a nice syntax for rdf list.

Reply:
I have made the other fixes. Great input.  As far as the query we need it the same as the one used in the MongoRyaDirectExample.  Maybe in the next version we can change it. Thank you.